warehouse/query-core/src/main/java/datawave/query/jexl/visitors/PushdownUnindexedFieldsVisitor.java

```diff
@@ -0,0 +1,131 @@
+package datawave.query.jexl.visitors;
+
+import static datawave.query.jexl.nodes.QueryPropertyMarker.MarkerType.BOUNDED_RANGE;
+import static datawave.query.jexl.nodes.QueryPropertyMarker.MarkerType.EVALUATION_ONLY;
+
+import java.util.Set;
+
+import org.apache.commons.jexl3.parser.ASTAndNode;
+import org.apache.commons.jexl3.parser.ASTEQNode;
+import org.apache.commons.jexl3.parser.ASTERNode;
+import org.apache.commons.jexl3.parser.ASTReferenceExpression;
+import org.apache.commons.jexl3.parser.JexlNode;
+import org.apache.commons.jexl3.parser.JexlNodes;
+import org.apache.log4j.Logger;
+
+import datawave.query.jexl.JexlASTHelper;
+import datawave.query.jexl.LiteralRange;
+import datawave.query.jexl.nodes.QueryPropertyMarker;
+
+/**
+ * Visitor meant to 'push down' predicates for expressions that are not executable against the global OR field index.
+ */
+public class PushdownUnindexedFieldsVisitor extends RebuildingVisitor {
+
+    private static final Logger log = Logger.getLogger(PushdownUnindexedFieldsVisitor.class);
+
+    protected Set<String> unindexedFields;
+
+    /**
+     * Construct the visitor
+     *
+     * @param unindexedFields
+     *            the fields being considered unindexed
+     */
+    public PushdownUnindexedFieldsVisitor(Set<String> unindexedFields) {
+        this.unindexedFields = unindexedFields;
+    }
+
+    /**
+     * helper method that constructs and applies the visitor.
+     *
+     * @param unindexedFields
+     *            the fields considered unindexed
+     * @param queryTree
+     *            the query tree
+     * @param <T>
+     *            type of the query tree
+     * @return a reference to the node
+     */
+    public static <T extends JexlNode> T pushdownPredicates(T queryTree, Set<String> unindexedFields) {
+        PushdownUnindexedFieldsVisitor visitor = new PushdownUnindexedFieldsVisitor(unindexedFields);
+        return (T) (queryTree.jjtAccept(visitor, null));
+    }
+
+    @Override
+    public Object visit(ASTAndNode node, Object data) {
+        // if not already delayed somehow
+        if (!QueryPropertyMarker.findInstance(node).isAnyTypeExcept(BOUNDED_RANGE)) {
+            LiteralRange range = JexlASTHelper.findRange().getRange(node);
```

Comment:
This can be simplified to `if(QueryPropertyMarker.findInstance(node).isType(BOUNDED_RANGE)) {`